File: Python/bestTimeToBuyAndSellAStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        bp = float('inf')
        max_profit = 0
        for price in prices:
            if price < bp:
                bp = price
            elif price - bp > max_profit:
                max_profit = price - bp
        return max_profit

File: Kotlin/bestTimeToBuyAndSellAStock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var bp = prices[0]
        var maxProfit = 0
        for (price in prices){
            if (price < bp){
                bp = price
            }else if(price - bp > maxProfit){
                maxProfit = price - bp
            }
        }
        return (maxProfit)
    }
}